src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -121,4 +124,53 @@ public static Set<Tag> parseTags(Collection<String> tags) throws ParseException
         }
         return tagSet;
     }
+
+    //=========== LeavesBook ================================================================================
+
+    /**
+     * Parses a {@code String title} into a {@code String}.
+     * Leading and trailing whitespaces will be trimmed.
+     */
+    public static String parseTitle(String title) {
+        requireNonNull(title);
+        String trimmedTitle = title.trim();
+        return trimmedTitle;
+    }
+
+    /**
+     * Parses a {@code String start} {@code String end} into an {@code ArrayList<Date>}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code start} and {@code end} is invalid.
+     */
+    public static ArrayList<Date> parseDate(String start, String end) throws ParseException {
+        requireNonNull(end);
+        requireNonNull(start);
+        String trimmedDateEnd = end.trim();
+        String trimmedDateStart = start.trim();
+        ArrayList<Date> validDate = new ArrayList<>();
```

Comment:
I think in future it might be safer to have a Range object because the danger of returning an ArrayList is that the size is not guaranteed, but I'll approve it anyway for now.